pages/index.js

```diff
@@ -39,12 +43,23 @@ export default function Home({
     closeOverlay();
   };
 
+  /////////////////tracking the number of checkedgifts
+  ////////////////update the number as soon as giftlist changes
+  const [numberCheckedGifts, setNumberCheckedGifts] = useState("");
+  useEffect(() => {
+    const numberCheckedGifts = giftList.filter((gift) => {
+      return gift.isChecked === true;
+    });
+    setNumberCheckedGifts(numberCheckedGifts.length);
+  }, [giftList]);
```

Comment:
This is a tricky one. I had the same thinking like you at the beginning. But, when you want to have just the length of the `numberCheckedGifts` everytime, you can omit the `state` and the `useEffect` and just save the length at every new render.

```suggestion
const checkedGifts = giftList.filter((gift) => gift.isChecked === true );
const numberCheckedGifts = checkedGifts.length
```

Reply:
cool, makes it more simple. I´ll use it as you suggested